src/torrent.cpp

```diff
@@ -1142,17 +1142,13 @@ namespace libtorrent
 #ifndef TORRENT_DISABLE_LOGGING
 		debug_log("*** set-share-mode: %d", s);
 #endif
-
-		// in share mode, all pieces have their priorities initialized to 0
-		if (m_share_mode && valid_metadata())
+		if (m_share_mode)
 		{
-			m_file_priority.clear();
-			m_file_priority.resize(m_torrent_file->num_files(), 0);
+			int const num_files = valid_metadata() ? m_torrent_file->num_files() : m_file_priority.size();
+			// in share mode, all pieces have their priorities initialized to 0
+			prioritize_files(std::vector<int>(num_files));
+			recalc_share_mode();
```

Comment:
this seems like a reasonable refactor, but unrelated to the issue, right?

Reply:
Each piece of code that deals with file priorities is presumably related.

Reply:
so the relevant change here is to call `prioritize_files()` regardless of whether we have metadata or not, right?

Reply:
Yes, I believe that priorities must be handled in one place, that is `torrent::prioritize_files`. 'Empty metadata' case is covered there.